### DISCUSSION TITLE
Can I replace a nested field of one-to-many relation with an array ?

### DISCUSSION THREAD

**Discussion by mraxime** [prisma/prisma#6347] [Q&A] [answered]
Hello friends, I have this model:

```prisma
model Step {
  id          Int        @id @default(autoincrement())
  title       String     @unique
  items       StepItem[]
}

model StepItem {
  id             Int     @id @default(autoincrement())
  title          String

  // Relation fileds
  Step   Step @relation(fields: [stepId], references: [id])
  stepId Int
}
```

which has a one-to-many relation. (In this scenario one step can have many stepItems)

and let's say I have an array with many stepItems:

```ts
const newArr = [
    {
      id: 1,
      title: 'title_1',
      stepId: 1,
    },
    {
      id: 2,
      title: 'title_2',
      stepId: 1,
    },
  ];
```

and using an operation similar to this:

```ts
await prisma.step.update({
      data: { id: 1, title: 'My_New_Title', items: { create: newArr } },
      where: { id: 1 },
    });
```

Can I just replace the items with my new array? Because if I let the above operation with a create, it won't work. I just want to "force" the items of the step, to be my `newArr`. Is it possible? Thank you !

**Comment by ryands17** [comment, marked as answer]
@Maxime50 👋 

To replace the items, you can do a `deleteMany` followed by a `create`:

```ts
await prisma.step.update({
      data: { id: 1, title: 'My_New_Title', items: { 
        deleteMany: {},
        create: newArr 
      }},
      where: { id: 1 },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
